backfill.py

```diff
@@ -130,27 +148,126 @@ def get_result_from_durations(start_up_durations_path, analyzed_path):
     analyze_durations.save_output(stats, analyzed_path)
 
 
-def run_performance_analysis_on_nightly(package_id, path_to_measure_start_up_script, array_of_apk_path):
-    for apk_path in array_of_apk_path:
-        analyze_nightly_for_one_build(package_id, path_to_measure_start_up_script, apk_path)
+def run_performance_analysis_on_nightly(package_id, path_to_measure_start_up_script, array_of_apk_path, build_type):
+    for idx, apk_path in enumerate(array_of_apk_path):
+        analyze_nightly_for_one_build(idx, package_id, path_to_measure_start_up_script, apk_path, build_type)
+
+
+def fetch_repository(repository_path, remote_name):
```

Comment:
thought (non-blocking): this method feels like it could integrate better with the whole. In a subsequent PR, let's...
- I think we should only do this if the caller provided the `remote_name` so we don't end up crashing if we guess the upstream is called `upstream` and it's not
- then in this case, if this call fails, I feel like we should stop the script. In the current code, we should gracefully handle that case which could make it more complicated.